hubblestack/hec/opt.py

```diff
@@ -69,7 +82,13 @@ def _get_splunk_options(space, modality, **kw):
             for k in opt:
                 j = nicknames.get(k, k)
                 if j in final_opts:
-                    final_opts[j] = opt[k]
+                    # if j is one of the args that can be overridden and has been provided then do not update it
+                    if j in {'token', 'index', 'port'}:
```

Comment:
You have not used 'options_for_grains_config' dict defined at the beginning. I guess you intended to use it here ?